### PR TITLE
Use Objects.requireNonNull().

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Use `Objects.requireNonNull()` to ensure null safety instead of throwing `NullPointerException`s manually.

#### APK testing 

[debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5732929/debug.zip)

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/us/shandian/giga/get/sqlite/FinishedMissionStore.java [hunk 1]
```diff
@@ -186,15 +186,14 @@ private FinishedMission getMissionFromCursor(Cursor cursor) {
     }
 
     public void addFinishedMission(DownloadMission downloadMission) {
-        if (downloadMission == null) throw new NullPointerException("downloadMission is null");
         SQLiteDatabase database = getWritableDatabase();
-        ContentValues values = getValuesOfMission(downloadMission);
+        ContentValues values = getValuesOfMission(Objects.requireNonNull(downloadMission,
+                "downloadMission is null"));
```
Comment (TobiGr) [inline app/src/main/java/us/shandian/giga/get/sqlite/FinishedMissionStore.java:191]: move this up

File: app/src/main/java/us/shandian/giga/get/sqlite/FinishedMissionStore.java [hunk 2]
```diff
@@ -212,9 +211,9 @@ public void deleteMission(Mission mission) {
     }
 
     public void updateMission(Mission mission) {
-        if (mission == null) throw new NullPointerException("mission is null");
         SQLiteDatabase database = getWritableDatabase();
-        ContentValues values = getValuesOfMission(mission);
+        ContentValues values = getValuesOfMission(Objects.requireNonNull(mission,
+                "mission is null"));
```
Comment (TobiGr) [inline app/src/main/java/us/shandian/giga/get/sqlite/FinishedMissionStore.java:216]: move this up

**Review by TobiGr — COMMENTED** [review]
Thanks. Changes look good.

I think it makes sense to have the `null` checks at the beginning of a method to not execute code unnecessarily.

**Comment by Stypox** [comment]
I'd also remove some strings which are `Objects.requireNonNull()` parameters, when they don't tell anything about what caused the issue that would be easily extracted from the exception. what do you think @TobiGr ?

**Comment by TobiGr** [comment]
@Stypox Sounds good

File: app/src/main/java/us/shandian/giga/get/DownloadMission.java [hunk 3]
```diff
@@ -154,8 +155,8 @@
     public transient Thread init = null;
 
     public DownloadMission(String[] urls, StoredFileHelper storage, char kind, Postprocessing psInstance) {
-        if (urls == null) throw new NullPointerException("urls is null");
-        if (urls.length < 1) throw new IllegalArgumentException("urls is empty");
+        if (Objects.requireNonNull(urls, "urls is null").length < 1)
```
Comment (B0pol) [inline app/src/main/java/us/shandian/giga/get/DownloadMission.java:158]: are

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragment.java [hunk 4]
```diff
@@ -35,12 +36,9 @@
     private final CompositeDisposable compositeDisposable = new CompositeDisposable();
 
     public static LicenseFragment newInstance(final SoftwareComponent[] softwareComponents) {
-        if (softwareComponents == null) {
-            throw new NullPointerException("softwareComponents is null");
-        }
         final LicenseFragment fragment = new LicenseFragment();
```
Comment (Redirion) [inline app/src/main/java/org/schabi/newpipe/about/LicenseFragment.java:39]: move this line below the requireNonNull to avoid code execution on null

**Review by Redirion — APPROVED** [review]

**Pull request merged**